js/gestionPresupuestoWeb.js

```diff
@@ -80,6 +105,95 @@ function mostrarGastoAgrupadosWeb(idElemento, agrup, periodo){
 
 }
 
+function repintar(){
+
+    mostrarDatoEnId("presupuesto", mostrarPresupuesto());
+    mostrarDatoEnId("gastos-totales", calcularTotalGastos());
+    mostrarDatoEnId("balance-total", calcularBalance());
+
+    document.getElementById("listado-gastos-completo").innerHTML = "";
+    let listagastos = listarGastos();
+
+    for (let lista of listagastos)
+    {
+        mostrarGastoWeb("listado-gastos-completo", lista);
+    }
+}
+
+function EditarHandle(){
+    this.handleEvent = function(e){
+
+        let nuevadesc = prompt("Introduce nueva descripción");
+        this.gasto.actualizarDescripcion(nuevadesc);
+
+        let nuevovalor = prompt("Introduce nuevo valor");
+        nuevovalor = parseFloat(nuevovalor);
+        this.gasto.actualizarValor(nuevovalor);
+
+        let nuevafecha = prompt("Introduce nueva fecha");
+        nuevafecha = Date.parse(nuevafecha);
+        this.gasto.actualizarFecha(nuevafecha);
+
+        let nuevaetiqueta = prompt("Introduce nuevas etiquetas");
+        this.gasto.etiquetas = nuevaetiqueta.split(', ');
+
+        repintar();
+        
+    }
+}
+
+function BorrarHandle(){
+    this.handleEvent = function(e){
+
+        borrarGasto(this.gasto.id);
+
+        repintar();
+    }
+}
+
+function BorrarEtiquetasHandle(){
+    this.handleEvent = function(e){
+
+        this.etiqueta = this.etiqueta.split(", ");
```

Comment:
No es necesario nada de esto... `this.etiqueta` es un string. Por tanto, puedes pasarlo como parámetro directamente a `borrarEtiquetas`. No tienes que convertirlo a array ni nada por el estilo.